desc/coils.py

```diff
@@ -560,6 +632,192 @@ def from_symmetry(cls, coils, NFP, sym=False):
 
         return cls(*coilset)
 
+    @classmethod
+    def from_makegrid_coilfile(cls, coil_file, method="cubic"):
+        """Create a CoilSet of SplineXYZCoils from a MAKEGRID-formatted coil txtfile.
+
+        Parameters
+        ----------
+        coil_file : str or path-like
+            path to coil file in txt format
+        method : str
+            method of interpolation
+            - `'nearest'`: nearest neighbor interpolation
+            - `'linear'`: linear interpolation
+            - `'cubic'`: C1 cubic splines (aka local splines)
+            - `'cubic2'`: C2 cubic splines (aka natural splines)
+            - `'catmull-rom'`: C1 cubic centripetal "tension" splines
+        """
+        coils = []  # list of SplineXYZCoils
+        coilinds = [2]  # always start at the 3rd line
+        names = []
+
+        # read in the coils file
+        with open(coil_file) as f:
+            lines = f.readlines()
+            for i, line in enumerate(lines):
+                if line.find("periods") != -1:
+                    NFP = int(line.split()[-1].strip())
+                    continue
+                if (
+                    line.find("begin filament") != -1
+                    or line.find("end") != -1
+                    or line.find("mirror") != -1
+                ):
+                    continue  # skip headers and last line
+                if (
+                    len(line.split()) != 4
+                    and line.strip()  # ensure not counting blank lines
+                ):
+                    coilinds.append(i)
+                    names.append(" ".join(line.split()[4:]))
+        if len(lines[3].split()) != 4:
+            raise OSError(
+                "4th line in file must be the start of the first coil!"
+                + "Expected a line of length 4 (after .split()),"
+                + f"instead got length {lines[3].split()}"
+            )
+        header_lines_not_as_expected = np.array(
+            [
+                len(lines[0].split()) != 2,
+                len(lines[1].split()) != 2,
+                len(lines[2].split()) != 2,
+            ]
+        )
+        if np.any(header_lines_not_as_expected):
+            raise OSError(
+                "first 3 lines in file must be the header lines,"
+                + " each of length 2 (after .split())!"
+                + f"Line(s) {lines[np.where(header_lines_not_as_expected)[0]]}"
+                + " is not length 2"
+            )
+
+        for i, (start, end) in enumerate(zip(coilinds[0:-1], coilinds[1:])):
+            coords = np.genfromtxt(lines[start + 1 : end])
+            if i % 20 == 0:
```

Comment:
is this assuming the coil only has 20 points?

Reply:
no, it is a holdout from debugging (making sure I knew the load function was working by printing progress), I will remove